src/app/courses/courses.service.ts

```diff
@@ -26,7 +27,7 @@ export class CoursesService {
   // If the id already matches what is stored on the service, return that.
   // Or will get new version if forceLatest set to true
   requestCourse({ courseId, forceLatest = false }, opts: any = {}) {
-    if (!forceLatest && this.course && courseId === this.course._id) {
+    if (!forceLatest && courseId === this.course._id) {
```

Comment:
So, I added a check for this.course because it's null sometimes and gives an error when accessing the id

Reply:
For example, if you directly go to the step view page then BehaviorSubject is null and throws an error